cdap-api/src/main/java/co/cask/cdap/api/schedule/Schedule.java

```diff
@@ -26,14 +26,47 @@
 
   private final String description;
 
+  // NOTE: the two below attributes are left for backwards compatibility
   private final String cronEntry;
 
+  /**
+   * Build a time-based schedule.
+   *
+   * @param name name of the schedule
+   * @param description description of the schedule
+   * @param cronExpression cron expression for the schedule
+   * @return a schedule based on the given {@code cronExpression}
+   */
+  public static Schedule buildTimeSchedule(String name, String description, String cronExpression) {
```

Comment:
The idea is to limit what users should interact with only to the ``Schedule`` class. They should never have to deal directly with ``TimeSchedule`` or ``StreamSizeSchedule`` classes, so that we can change them later without breaking compatibility.
That also means that I should remove the APIs of ``AbstractApplication`` that accept ``TimeSchedule`` or ``StreamSizeSchedule`` objects.